sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -322,6 +333,9 @@ class Dataset[T] private[sql](
   }
 
   override def toString: String = {
+    if (name != null) {
+      return name
+    }
```

Comment:
The storage tab gets it from the cache manager (which gets it via Dataset.persist()). I think this is what will be displayed when the Dataset is printed in the repl, I'm not sure if it's also used elsewhere.
Given that there's also a .name method I agree that it would be nice to keep the schema in here. I'll add it in, but will need to update the tests to take that into account.